src/nv_ingest/modules/transforms/text_splitter.py

```diff
@@ -0,0 +1,171 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+import copy
+import logging
+import traceback
+import uuid
+from typing import Any
+from typing import List
+
+import mrc
+import pandas as pd
+from transformers import AutoTokenizer
+from morpheus.messages import ControlMessage
+from morpheus.messages import MessageMeta
+from morpheus.utils.control_message_utils import cm_skip_processing_if_failed
+from morpheus.utils.module_utils import ModuleLoaderFactory
+from morpheus.utils.module_utils import register_module
+from mrc.core import operators as ops
+from pydantic import BaseModel
+
+import cudf
+
+from nv_ingest.schemas.metadata_schema import ContentTypeEnum
+from nv_ingest.schemas.text_splitter_schema import TextSplitterSchema
+from nv_ingest.util.exception_handlers.decorators import nv_ingest_node_failure_context_manager
+from nv_ingest.util.flow_control import filter_by_task
+from nv_ingest.util.modules.config_validator import fetch_and_validate_module_config
+from nv_ingest.util.tracing import traceable
+
+logger = logging.getLogger(__name__)
+
+
+def _build_split_documents(row, chunks: List[str]) -> List[dict[str, Any]]:
+    """Build documents from text chunks"""
+    documents: List[dict] = []
+
+    for i, text in enumerate(chunks):
+        if text is None or not text.strip():
+            continue
+
+        metadata = row.metadata if hasattr(row, "metadata") and isinstance(row.metadata, dict) else {}
+        metadata = copy.deepcopy(metadata)
+
+        metadata["content"] = text
+
+        documents.append({"document_type": ContentTypeEnum.TEXT.value, "metadata": metadata, "uuid": str(uuid.uuid4())})
+
+    return documents
+
+
+def _split_into_chunks(text, tokenizer, chunk_size=300, chunk_overlap=0):
+    # Tokenize the text into token IDs
+    encoding = tokenizer.encode_plus(text, add_special_tokens=False, return_offsets_mapping=True)
+
+    # Get the token IDs and offsets for splitting
+    tokens = encoding["input_ids"]
+    offsets = encoding["offset_mapping"]
+
+    # Split the tokens into chunks of the desired size
+    chunks = [tokens[i : i + chunk_size] for i in range(0, len(tokens), chunk_size - chunk_overlap)]
```

Comment:
Related to my comment in the schema, we will want to be sure that chunk_overlap is strictly less than chunk_size. Otherwise we could conceivably get a negative size here.